test/test_databases/test_bigquery.py

```diff
@@ -338,8 +338,9 @@ def test_copy(self):
         gcs_client = self._build_mock_cloud_storage_client(tmp_blob_uri)
         tbl = self.default_table
         bq = self._build_mock_client_for_copying(table_exists=False)
-        bq.copy_from_gcs = mock.MagicMock()
-        table_name = ("dataset.table",)
+        bq._load_table_from_uri = mock.MagicMock()
```

Comment:
To be honest, I'm not sure if these unit tests have any value. I'm a big fan of tests, and even TDD where it makes sense, but some code just doesn't have any testable surface area. (Outside of integration tests that do actually hit the external service.)

There's a few code smells here like mocking private methods, and asserting that private methods were called a certain number of times.

This might be too big of a change for this commit, but my vote would be that we just scrap these tests.

Reply:
At the very least, I think a better long term direction for these tests would be to mock out the `client` object that the BQ connector holds on to, but _not_ to mock the private methods of the BQ connector itself.

Reply:
(Again, I don't think that's worth changing in this PR, though.)

Reply:
As an aside, this is an endemic problem for many parsons tests. There many tests that are essentially
```
Foo.bar = mock.MagicMock()
Foo.bar.return_value = 'fibble'
result = Foo.bar()
assert result == 'fibble'
```

Reply:
Yeah, more than willing to rethink how we do tests, but don't let it get in the way of this particular PR.

Reply:
Yeah, I guess that Will should probably just leave this as-is in this PR. More of a philosophical point, then.